tests/rados/test_reads_balancer.py

```diff
@@ -278,8 +282,175 @@ def run(ceph_cluster, **kw):
+                        log.error(
+                            f"failed to generate read balancer file at loc : {read_file_loc}"
+                        )
+                        raise Exception("read balancer file not generated")
+
+                    log.debug(f"read balancer file generated at loc : {read_file_loc}")
+
+                    # Checking the state of pool before applying the reads balancing
+                    disallowed_states = [
+                        "remapped",
+                        "degraded",
+                        "incomplete",
+                        "undersized",
+                    ]
+                    clean_pgs = rados_obj.check_pool_pg_states(
+                        pool=pool_name, disallowed_states=disallowed_states
+                    )
+
+                    # Executing the read results on the cluster.
+                    try:
+                        client_node.exec_command(
+                            sudo=True, cmd=f"source {read_file_loc}"
+                        )
+                    except Exception as e:
+                        log.error(
+                            f"Exception hit during setting the read recommendations. {e}"
+                        )
+                        log.debug(
+                            "The exception hit is expected due to bug :"
+                            " https://bugzilla.redhat.com/show_bug.cgi?id=2237574."
+                            " The below exception to be uncommented post bug fix."
+                        )
+                        # raise Exception("read recommendations not set")
+
+                    log.info(
+                        f"Successfully set the read recommendations on the pool : {pool_name}"
+                    )
+                    time.sleep(5)
+                    # There should be no data movement b/w the PGs when read balancing was performed.
+                    # Checking the same if we had clean pgs to start with
+                    if clean_pgs:
+                        if not rados_obj.check_pool_pg_states(
+                            pool=pool_name, disallowed_states=disallowed_states
+                        ):
+                            log.error(
+                                "All the PGs are not active + clean post addition of reads balancing"
+                                f"There should be no data movement. Observed with pool {pool_name}"
+                            )
+                            raise Exception("Data movement post reads balancing error")
+                        log.debug(f"Clean PGs on pool {pool_name} post reads balancing")
+
+                    log.debug("Sleeping for 20 seconds for scores to be updated")
+                    time.sleep(20)
+                    pool_details = rados_obj.get_pool_details(pool=pool_name)
+                    final_read_scores = pool_details["read_balance"]
+                    final_score = init_read_scores["score_acting"]
```

Comment:
final_score = **final_read_scores**["score_acting"]

Reply:
Thanks for the catch! good one!